Bottom: 135d82583dcafb485a79870588dcb7758be0fc8d
Top:    97a8299f1e7121b419d70cbf392fe3353aa83072
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-08 01:58:00 +0000

Add profiling to file load













---

diff --git a/src/file.c b/src/file.c
index 738188a..35bfb57 100644
--- a/src/file.c
+++ b/src/file.c
@@ -350,6 +350,10 @@ LoadPCB (char *Filename)
 {
   PCBTypePtr newPCB = CreateNewPCB (False);
   Boolean units_mm;
+  clock_t start, end;
+  double elapsed;
+
+  start = clock();
 
   /* new data isn't added to the undo list */
   if (!ParsePCB (newPCB, Filename))
@@ -380,6 +384,10 @@ LoadPCB (char *Filename)
 	  CreateDefaultFont ();
 	}
 
+  end = clock();
+  elapsed = ((double) (end - start)) / CLOCKS_PER_SEC;
+  printf ("Loading file %s took %f\n", Filename, elapsed);
+
       /* clear 'changed flag' */
       SetChangedFlag (False);
       PCB->Filename = MyStrdup (Filename, "LoadPCB()");
